--- a/src/rateup.c
+++ b/src/rateup.c
@@ -1452,12 +1452,7 @@ update (in, out, abs_max, absupdate)
 	    {
 	      inr = diff (in, last.in);
 	      if (inr < 0) {
-                if (inr > - (long long) 1 << 32) { 	/* wrapped 32-bit counter? */
-  		    inr += (long long) 1 << 32;
-                }
-                else {
-  		    inr = 0;
-                }
+  		inr = 0;
               }                                        
 	    }
 	  if (absupdate == 2)
@@ -1498,13 +1493,8 @@ update (in, out, abs_max, absupdate)
 	  else
 	    {
 	      outr = diff (out, last.out);
-	      if (outr < 0) {	/* wrapped  counter? */
-                if (outr > - (long long) 1 << 32) {
-  		    outr += (long long) 1 << 32;
-                }
-                else {
-  		    outr = 0; /* 64bit counters do not wrap usually */
-                }
+	      if (outr < 0) {	
+  		outr = 0; 
               }
 	    }
 	  if (absupdate == 2)
